### ISSUE TITLE
Support for Data Source - azurerm_application_gateway

### ISSUE THREAD

**Issue by kgopi1** [hashicorp/terraform-provider-azurerm#10503]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### New or Affected Resource(s)

* azurerm_application_gateway. 
Data source provider is not available for  azurerm_application_gateway , can you provide the data resource for this provider. 

### Potential Terraform Configuration

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key.
```

### References

* #0000

**Comment by brunhil** [comment]
Hi @kgopi1 , I'm taking a look at this at the moment. Do you have any request for specific properties to include? There are quite a for the AppGateway resource and most seem configurable for create. I'm thinking `identity` would be useful to include.

**Comment by SnuhNZ** [comment]
Hi Guys,
Yeah this would be handy for us too, we're setting up a project where we want one App GW and the ability to add and remove listeners/rules/http settings/backends to the existing gateway for small slices of environments. i.e. if we had a data source for the app gateway and then could request a listener or a rule etc.

Cheers

**Comment by brunhil** [comment]
Looks like this was merged with #10268. Any chance this can be closed as well @tombuildsstuff ?

**Comment by kgopi1** [comment]
Thanks for providing Data source for Application gateway , but still we cant use it , getting below error . 

data.azurerm_application_gateway.example: Refreshing state...

Error: Invalid address to set: []string{"identity", "0", "principal_id"}

  on input.tf line 76, in data "azurerm_application_gateway" "example":
  76: data "azurerm_application_gateway" "example" {

**Comment by max77p** [comment]
This is affecting us as well, cannot even use the example as is

**Comment by RustyF** [comment]
Still present in 2.56, btw

**Comment by jakobyte1024** [comment]
Unfortunately still present in 2.59

**Comment by LoicAG** [comment]
@brunhil I was also able to reproduce; the error only shows up on an Application Gateway on which an `identity` was defined, it works fine if `identity` is `null`.

Extract of the output of `az network application-gateway show -n name -g group`:

This is KO:
```
  "id": "/subscriptions/.../resourceGroups/../provider
s/Microsoft.Network/applicationGateways/...",
  "identity": {
    "principalId": null,
    "tenantId": null,
    "type": "userAssigned",
    "userAssignedIdentities": {
      "/subscriptions/.../resourcegroups/.../providers/Microsoft.ManagedIdentity/userAssignedIdentities/...": {
        "clientId": "...",
        "principalId": "..."
      }
    }
  },
```

This is OK:
```
  "id": "/subscriptions/.../resourceGroups/.../providers
/Microsoft.Network/applicationGateways/...",
  "identity": null,
```

**Comment by tombuildsstuff** [comment]
Fixed via #12226
